### ISSUE TITLE
Can i train mml-lgb model and loading in local Python?

### ISSUE THREAD

**Issue by RainFung** [Azure/mmlspark#1084]
Code like below
```
from mmlspark.lightgbm import LightGBMClassifier
model =LightGBMClassifier(boostingType='gbdt',
                          objective='binary',
                          baggingFreq=5,
                          learningRate=0.1,
                          numIterations=50,
                          earlyStoppingRound=10,
                          featuresCol = 'features',
                          categoricalSlotNames = cate_list,
                          numLeaves=30,
                          labelCol="label").fit(train)

model.saveNativeModel(path)

from sklearn.externals import joblib
clf = joblib.load(path) 
```

**Comment by imatiach-msft** [comment]
@RainFung yes, that should work, are you seeing any errors with that code?  If so, maybe it's something about native lightgbm differing between python vs mmlspark versions, but I thought the format is backwards compatible.

**Comment by imatiach-msft** [comment]
one thing, I think you can't call this:
from sklearn.externals import joblib
clf = joblib.load(path) 

you need to load using the lightgbm python api

**Comment by imatiach-msft** [comment]
you can create the python booster from model str:
https://github.com/microsoft/LightGBM/issues/2097
model = lgb.Booster({'model_str': buf.read().decode("UTF-8")})
you can also create the scikit-learn API but it's not "officially supported":
https://github.com/microsoft/LightGBM/issues/1942

**Comment by RainFung** [comment]
I havent try yet.
My plan is :
1.  train data by `mmlspark lightgbm`
2. save model via `saveNativeModel`
3. get model from hdfs path
4. load model via `joblib`
5. predict pandas dafaframe via joblib model

Can you give me some demo?

**Comment by imatiach-msft** [comment]
yes, that last part: "load model via joblib"
you can't load it via joblib.  It's just a file, it's the lightgbm native representation of the booster.  It is readable.  You can load it in a python booster.  I don't have any specific demo but I can help write the code.

**Comment by RainFung** [comment]
Thanks. I'll follow your step using `lgb booster`.

**Comment by RainFung** [comment]
> you can create the python booster from model str:
> [microsoft/LightGBM#2097](https://github.com/microsoft/LightGBM/issues/2097)
> model = lgb.Booster({'model_str': buf.read().decode("UTF-8")})
> you can also create the scikit-learn API but it's not "officially supported":
> [microsoft/LightGBM#1942](https://github.com/microsoft/LightGBM/issues/1942)

What are the advantages compared to file and model string?

**Comment by imatiach-msft** [comment]
@RainFung there shouldn't be any advantage/disadvantage, the string itself is the same, they are just different methods to get the model string

**Comment by RainFung** [comment]
Thanks，I finished the project alreday.
